pkg/networkserver/mac/adr.go

```diff
@@ -78,14 +78,99 @@ const (
 	DefaultADRMargin = 15
 )
 
-func DeviceADRMargin(dev *ttnpb.EndDevice, defaults ttnpb.MACSettings) float32 {
-	if v := dev.GetMacSettings().GetAdrMargin(); v != nil {
-		return v.Value
+func deviceUseADR(dev *ttnpb.EndDevice, defaults ttnpb.MACSettings, phy *band.Band) bool {
+	switch {
+	case dev.GetMulticast():
+		return false
+
+	case dev.GetMacSettings().GetAdr().GetDisabled() != nil:
+		return false
+	case dev.GetMacSettings().GetAdr().GetStatic() != nil:
+		return true
+
+	case defaults.Adr.GetDisabled() != nil:
+		return false
+	case defaults.Adr.GetStatic() != nil:
+		return true
+
+	case !phy.EnableADR:
+		return false
```

Comment:
Shouldn't `EnableADR` be renamed to `SupportsADR`? or even `SupportsDynamicADR`?